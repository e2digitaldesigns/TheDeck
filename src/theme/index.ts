export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
