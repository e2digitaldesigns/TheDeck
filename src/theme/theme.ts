export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
