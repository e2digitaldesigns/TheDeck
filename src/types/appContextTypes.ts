export interface IntAppDataIconSelector {
  isVisible: boolean;
  icon: string;
}

export interface IntAppDataActive {
  profileId: string;
  pageId: string;
  buttonPadId: string;
  actionId: string;
}

export interface IntDropZones {
  actionList: boolean;
  buttonPads: boolean;
  sideBarProfiles: boolean;
  sideBarStyles: boolean;
  styleHeader: boolean;
}

export interface IntAppContextInterface {
  iconSelector: IntAppDataIconSelector;
  active: IntAppDataActive;
  dropZones: IntDropZones;
}

export interface IntAppContextStateInterface {
  appState: IntAppContextInterface;
  setAppState: React.Dispatch<React.SetStateAction<any>>;
}

export type IntAppData = IntAppContextStateInterface;

export enum AppStateActive {
  ProfileId = "profileId",
  PageId = "pageId",
  ButtonPadId = "buttonPadId",
  ActionId = "actionId"
}
