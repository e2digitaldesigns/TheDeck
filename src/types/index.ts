/* istanbul ignore file */
export * from "./actionType";
export * from "./aplicationTypes";
export * from "./appContextTypes";
export * from "./buttonPadGridTypes";
export * from "./dragAndDropTypes";
export * from "./globalContextType";
export * from "./globalType";
export * from "./ipcRenderTypes";
export * from "./obsTypes";
