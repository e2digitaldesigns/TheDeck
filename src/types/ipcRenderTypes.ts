export enum IpcRendererTypes {
  database = "database",
  databaseReturn = "database:return",
  header = "header",
  importSettings = "importSettings",
  importSettingsReturn = "importSettings:return",
  mdPlaySound = "MD:playSound"
}

export enum IpcRendererTypesAction {
  close = "close",
  fsToggle = "fsToggle",
  minimize = "minimize",
  loadAppData = "loadAppData",
  saveAppData = "saveAppData"
}
