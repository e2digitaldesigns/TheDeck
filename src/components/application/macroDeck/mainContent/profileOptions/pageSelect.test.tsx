import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import PageSelect from "./pageSelect";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../../../theme/theme";
import initGlobalState from "../../../../../__mock_data__/globalState.json";
import initAppState from "../../../../../__mock_data__/appState.json";
import _cloneDeep from "lodash/cloneDeep";

import { IntAppData, IntGlobalData } from "../../../../../types";

const mockGlobalState: IntGlobalData = {
  state: _cloneDeep(initGlobalState),
  setState: jest.fn(state => state)
};

const mockAppState: IntAppData = {
  appState: initAppState,
  setAppState: jest.fn(appState => appState)
};

const mocked = {
  usePage: {
    activatePage: jest.fn(),
    createPage: jest.fn(),
    deletePage: jest.fn(),
    readPages: initGlobalState.pages
  }
};

jest.mock("../../../../../hooks", () => ({
  useAppData: () => mockAppState,
  useGlobalData: () => mockGlobalState,
  usePage: () => ({
    activatePage: mocked.usePage.activatePage,
    createPage: mocked.usePage.createPage,
    deletePage: mocked.usePage.deletePage,
    readPages: () => ({ ...mocked.usePage.readPages })
  })
}));

const testSetup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <PageSelect />
    </ThemeProvider>
  );
};

let wrapper: any = null;

afterEach(() => {
  wrapper = null;
  jest.clearAllMocks();
});

describe("<PageSelect/>", () => {
  it("Should render without errors", () => {
    const component = testSetup();
    expect(component).toMatchSnapshot();
  });

  it("Should change page", () => {
    const component = testSetup();
    const select = component.getByTestId("page_select__select");
    expect(select).toBeTruthy();

    const page = mockGlobalState.state.pages[2];
    fireEvent.change(select, { target: { value: page._id } });
    expect(mocked.usePage.activatePage).toHaveBeenCalledTimes(1);
    expect(mocked.usePage.activatePage).toHaveBeenCalledWith(page._id);
  });

  it("Should delete page", () => {
    mockAppState.appState.active.pageId = mockGlobalState.state.pages[2]._id;
    const component = testSetup();
    const btn = component.getByTestId("page_select__remove");
    expect(btn).toBeTruthy();

    fireEvent.click(btn);
    expect(mocked.usePage.deletePage).toHaveBeenCalledTimes(1);
  });

  it("Should close icon page", () => {
    mockAppState.appState.iconSelector.isVisible = true;
    mockAppState.appState.active.pageId = mockGlobalState.state.pages[2]._id;
    const component = testSetup();
    const btn = component.getByTestId("page_select__close_icon");
    expect(btn).toBeTruthy();

    fireEvent.click(btn);
    expect(mockAppState.setAppState).toHaveBeenCalledTimes(1);
  });
});
