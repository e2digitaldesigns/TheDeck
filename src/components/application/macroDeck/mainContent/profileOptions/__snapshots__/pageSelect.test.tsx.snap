// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageSelect/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        
        <button
          class="sc-hKwDye sc-gKclnd PLLBx iQnRQN"
        >
          New Page
        </button>
        <select
          class="sc-dkPtRN sc-jRQBWg iRRDCl jxmNFY"
          data-testid="page_select__select"
        >
          <option
            value="5ed5ff96-38d4-4a68-a0b8-d8d7dbf7360f"
          >
            Page: 01
          </option>
          <option
            value="a14b3dd6-2468-4ec0-9c6c-acc52db3472a"
          >
            Page: 11
          </option>
          <option
            value="85e6c415-845b-4076-9aef-cac7e6b341ed"
          >
            Page: 21
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      
      <button
        class="sc-hKwDye sc-gKclnd PLLBx iQnRQN"
      >
        New Page
      </button>
      <select
        class="sc-dkPtRN sc-jRQBWg iRRDCl jxmNFY"
        data-testid="page_select__select"
      >
        <option
          value="5ed5ff96-38d4-4a68-a0b8-d8d7dbf7360f"
        >
          Page: 01
        </option>
        <option
          value="a14b3dd6-2468-4ec0-9c6c-acc52db3472a"
        >
          Page: 11
        </option>
        <option
          value="85e6c415-845b-4076-9aef-cac7e6b341ed"
        >
          Page: 21
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
